FrontEnd/src/Components/Profile/StudentSection.jsx

```diff
@@ -0,0 +1,177 @@
+  };
+
+  return (
+    <>
+      <Box
+        sx={{
+          pt: 2,
+          display: user?.p99roles.length === 0 ? 'flex' : 'none',
+          justifyContent: 'space-between',
+          mt: 2,
+        }}
+      >
+        <Typography variant="h6">Estudiante(s)</Typography>
+      </Box>
+      <Box>
+        {students !== null
+        && students.length === 0
+        && user?.p99roles.length === 0 ? (
+          <Box
+            sx={{
+              fontFamily: 'default',
+              fontSize: 'h3.fontSize',
+              py: 2,
+              display: 'flex',
+            }}
+          >
+            Registra un estudiante para poder inscribir clases!
+          </Box>
+          ) : (
+            students.map((student) => (
+              <StudentItem
+                key={student._id}
+                name={student.nombre}
+                studentInfo={student}
+                first_lastname={student.apellido_paterno}
+                second_lastname={student.apellido_materno}
+                editStudent={handleEditStudent}
+                setCurrentStudent={setCurrentStudent}
+                handleOpenDialog={handleOpenDialog}
+                setIsEditing={setIsEditing}
+              />
+            ))
+          )}
+        <Box
+          sx={{
+            display: {
+              xs: 'none',
+              md: user?.p99roles.length === 0 ? 'flex' : 'none',
+              justifyContent: 'flex-end',
+            },
+          }}
+        >
+          <Button
+            sx={{
+              textTransform: 'none',
+              backgroundColor: '#57a1f1',
+              fontSize: '18px',
+            }}
+            onClick={() => {
+              setAddStudent(!addStudent);
+            }}
+            variant="contained"
+            endIcon={<AddIcon />}
+          >
+            Agregar estudiante
+          </Button>
+        </Box>
+      </Box>
+
+      <Modal
+        open={addStudent}
+        onClose={() => setAddStudent(!addStudent)}
+        sx={{ overflow: 'scroll' }}
+      >
+        <>
+          <StudentProfile
+            studentInfo={defaultStudentInfo}
+            setAddStudent={setAddStudent}
+            addStudent={addStudent}
+            userID={user.sub}
+            setStudents={setStudents}
+            setSuccessOpen={setSuccessOpen}
+            setErrorOpen={setErrorOpen}
+            setAlertMessage={setAlertMessage}
+            setInfoOpen={setInfoOpen}
+          />
+        </>
+      </Modal>
+
+      <Modal
+        open={openEditModal}
+        onClose={() => setOpenEditModal(!openEditModal)}
+        sx={{ overflow: 'scroll' }}
+      >
+        <>
+          <EditStudentProfile
+            openEditModal={openEditModal}
+            setOpenEditModal={setOpenEditModal}
+            studentInfo={currentStudent}
+            setIsEditing={setIsEditing}
+            isEditing={isEditing}
+            setStudents={setStudents}
+            setSuccessOpen={setSuccessOpen}
```

Comment:
@TeraBlitz These components (`StudentProfile` and `EditStudentProfile`) are pretty much similar, I will try to use them as one component and I could save one component and also make `StudentSection` a little bit smaller.

Reply:
I will add it on the next PR.

Reply:
sounds good